src/alcotest/core.ml

```diff
@@ -345,7 +345,9 @@ module Make (M : Monad.S) = struct
     let open Suite in
     let test = suite.fn in
     let pp_event = pp_event t in
+    M.return () >>= fun () ->
```

Comment:
Why is this needed?

Reply:
This ensures that the printing of the event occurs inside the concurrency monad (which wouldn't be the case otherwise), which is necessary for the Alcotest output to be correctly interleaved with the test output when using `Alcotest_lwt`. It's not tested to be necessary, but it could be.

Reply:
Do you mind adding the test which shows that it is necessary? (I don't understand how the alcotest outputs can occur inside the Lwt monad.) There aren't any tests using alcotest_lwt and that expose this bug anyway, it is maybe not a bad idea to add one anyway.

Reply:
Sure, can do :slightly_smiling_face:

I added a test of the previous (broken) state in https://github.com/mirage/alcotest/pull/228/commits/30b47a7aa5a9bfb6a247288a0658895a777d4e21 (with an explanation). Note that the prints are observable but the `assert false` is not: the exception has been caught and hidden behind a bind point, so the failure isn't shown until the computation is run.

I fixed the behaviour in https://github.com/mirage/alcotest/pull/228/commits/847f2271a4cb99bb893189ff030b587b51c8076b. I went a step further and pushed the initial `"Testing ..."` message inside the concurrency monad too, for consistency.